src/content/posts/whats-possible-with-melange.mdx

```diff
@@ -0,0 +1,202 @@
+KeSearchBar_Css.re      KeSearchBar.re                  KeSearchVolume.re           KeSearchVolumeChart.re
+KeSearchVolumeGoogle.re KeSerpFeatures.re               KeSerpOverview_Css.re       KeSerpOverview.re
+KeSideMenu_Css.re       KeSideMenu.re                   KeSideMenuKlists.re         KeTablePagination_Css.re
+KeTablePagination.re    KeTrafficTable.re               KeTrafficTableRow_Css.re    KeTrafficTableRow.re
+KeUpdateButton.re       KeWidget_Css.re                 KeWidget.re                 KeWidgetChartCaption.re
+KeWidgetContent_Css.re  KeWidgetContent.re              KeWidgetHeader_Css.re       KeWidgetHeader.re
+KeWidgetPlaceholder.re  KeWidgetPlaceholder_Css.re      KeWidgetPreloader_Css.re    KeWidgetPreloader.re
+```
+
+The unique namespace works great for small applications, but when your codebase is big it becomes quite challenging to manage and not lose your hair. Aside, all dependencies/ppx's were applied to the entire monorepo, which made no-sense and caused issues all around.
+
+Since Melange, each app can drop the prefix and started to modularise each sub-application with logical boundaries instead, making sure each set of dependencies and ppx's are set correctly.
+
+## Enables universal code
+
+When I say “universal code” I refer to code that’s able to compile to both native and JavaScript. It’s useful for fullstack Reason development and a variety of use-cases, such as:
+
+- Type-safe data across client/server
+- Universal routing with type-safe routes, links and parameters
+- Re-use the same validation logic
+- Server rendering via server-reason-react
+- RPC (Remote Procedure Call) approach
+
+This has been one of the promises by Reason (and BuckleScript) since the beginning and it never materialised. I have seen some forms of this, but BuckleScript and dune never played well together.
+
+Now dune supports a new `melange` mode available for libraries, alongside with `native` mode, enabling universal code.
+
+Given a library without any system dependency (C bindings, unix module, etc) you are able to use it to emit JavaScript and run as an executable.
+
+Some parts of this are still on the design phase, but some more documentation has been written in the server-reason-react documentation:
+[How to organise the universal code](https://ml-in-barcelona.github.io/server-reason-react/local/server-reason-react/code-structure.html) &
+[Make sure your code is universal](https://ml-in-barcelona.github.io/server-reason-react/local/server-reason-react/universal-code.html).
+
+![Example of a react component rendered with both reason-react and server-reason-react](/images/server-reason-react-graph.png)
+
+## Preprocessing via ppx becomes better and faster
+
+ReScript doesn’t recommend the usage of ppx's and has been looking to replace them since the split. His claims are valid: they are a blackbox, are limited, slows the compilation and are hard to implement right.
```

Comment:
```suggestion
ReScript doesn’t recommend the usage of ppx's and has been looking to replace them since the split. Their reasoning is valid: they are a blackbox, are limited, slows the compilation and are hard to implement right.
```